api/tasks.go

```diff
@@ -1151,8 +1151,9 @@ func (t *TaskCSIPluginConfig) Canonicalize() {
 // WorkloadIdentity is the jobspec block which determines if and how a workload
 // identity is exposed to tasks.
 type WorkloadIdentity struct {
-	Name     string   `hcl:"name,optional"`
-	Audience []string `mapstructure:"aud" hcl:"aud,optional"`
-	Env      bool     `hcl:"env,optional"`
-	File     bool     `hcl:"file,optional"`
+	Name         string   `hcl:"name,optional"`
+	Audience     []string `mapstructure:"aud" hcl:"aud,optional"`
+	Env          bool     `hcl:"env,optional"`
+	File         bool     `hcl:"file,optional"`
+	NomadService string   `hcl:"nomad_service,optional"`
```

Comment:
We didn't get into this detail in the RFC but I wonder if calling this field name `NomadService` is potentially confusing for API consumers? At least for 1.7.0 this is always going to be a Consul service. We're going to write a claim `nomad_service` in the JWT but that's so that we're namespacing the JWT claim properly for Consul.

```suggestion
	ServiceName string   `hcl:"service_name,optional"`
```

Reply:
That can work for the `api` package but in `structs` we'd end up having to add extra handling for the signing code to take multiple types.

Reply:
@tgross I'll rename the field to `ServiceName` as you suggest but I'll keep the `WorkloadIdentity` struct.